### PR TITLE
Improve mnist examples and fix bugs.

### PR DESCRIPTION
**Pull request by zsdonghao**

### CODE REVIEW

**Comment by zsdonghao** [comment]
@luomai this commit do 2 things:

- 1. Simplified MNIST example
- 2. Fixed bug of D_TYPE (after we splited the `layers.py` into many files, `tutorial_minst_float16.py` have bug, as the `D_TYPE` can't be changed in `convolution.py` and others)

File: tensorlayer/layers/core.py [hunk 1]
```diff
@@ -8,17 +8,14 @@
...
-# __all__ = [
-#     "Layer",
-#     "DenseLayer",
-# ]
 
-# set_keep = locals()
-set_keep = globals()
-set_keep['_layers_name_list'] = []
-set_keep['name_reuse'] = False
+class LayersConfig:
```
Comment (lgarithm) [inline tensorlayer/layers/core.py]: Why not use instances of LayersConfig?

Comment (lgarithm) [reply tensorlayer/layers/core.py]:
```
class LayersConfig(object):
   def __init__(self):
      self.tf_dtype = tf.float32
      self. object_store = 
...


layers_config = LayersConfig()  # the default layer config.
```

Comment (lgarithm) [inline tensorlayer/layers/core.py]: class should derive from `object`:
```
class LayersConfig(object):
...
```

**Pull request merged**